src/CGALHybridPolyhedron.cc

```diff
@@ -0,0 +1,375 @@
+									}))
+			return;
+	}
+
+	nefPolyBinOp("nef difference", other,
+							 [&](nef_polyhedron_t &destinationNef, nef_polyhedron_t &otherNef) {
+								 CGALUtils::inPlaceNefDifference(destinationNef, otherNef);
+							 });
+}
+
+void CGALHybridPolyhedron::minkowski(CGALHybridPolyhedron &other)
+{
+	nefPolyBinOp("minkowski", other,
+							 [&](nef_polyhedron_t &destinationNef, nef_polyhedron_t &otherNef) {
+								 CGALUtils::inPlaceNefMinkowski(destinationNef, otherNef);
+							 });
+}
+
+void CGALHybridPolyhedron::transform(const Transform3d &mat)
+{
+	if (mat.matrix().determinant() == 0) {
+		LOG(message_group::Warning, Location::NONE, "", "Scaling a 3D object with 0 - removing object");
+		clear();
+	}
+	else {
+		auto t = CGALUtils::createAffineTransformFromMatrix<CGAL_HybridKernel3>(mat);
+
+		if (auto mesh = getMesh()) {
+			CGALUtils::transform(*mesh, mat);
+			cleanupMesh(*mesh, /* is_corefinement_result */ false);
+		}
+		else if (auto nef = getNefPolyhedron()) {
+			CGALUtils::transform(*nef, mat);
+		}
+		else {
+			assert(!"Bad hybrid polyhedron state");
+		}
+	}
+}
+
+void CGALHybridPolyhedron::resize(
+	const Vector3d &newsize, const Eigen::Matrix<bool, 3, 1> &autosize)
+{
+	if (this->isEmpty()) return;
+
+	transform(
+			CGALUtils::computeResizeTransform(getExactBoundingBox(), getDimension(), newsize, autosize));
+}
+
+CGALHybridPolyhedron::bbox_t CGALHybridPolyhedron::getExactBoundingBox() const
+{
+	bbox_t result(0, 0, 0, 0, 0, 0);
+	std::vector<point_t> points;
+	// TODO(ochafik): Optimize this!
+	foreachVertexUntilTrue([&](const auto &pt) {
+		points.push_back(pt);
+		return false;
+	});
+	if (points.size()) CGAL::bounding_box(points.begin(), points.end());
+	return result;
+}
+
+std::string CGALHybridPolyhedron::dump() const
+{
+	assert(!"TODO: implement CGALHybridPolyhedron::dump!");
+	return "?";
+	// return OpenSCAD::dump_svg(toPolySet());
+}
+
+size_t CGALHybridPolyhedron::memsize() const
+{
+	size_t total = sizeof(CGALHybridPolyhedron);
+	if (auto mesh = getMesh()) {
+		total += numFacets() * 3 * sizeof(size_t);
+		total += numVertices() * sizeof(point_t);
+	}
+	else if (auto nef = getNefPolyhedron()) {
+		total += nef->bytes();
+	}
+	return total;
+}
+
+void CGALHybridPolyhedron::foreachVertexUntilTrue(
+		const std::function<bool(const point_t &pt)> &f) const
+{
+	if (auto mesh = getMesh()) {
+		for (auto v : mesh->vertices()) {
+			if (f(mesh->point(v))) return;
+		}
+	}
+	else if (auto nef = getNefPolyhedron()) {
+		nef_polyhedron_t::Vertex_const_iterator vi;
+		CGAL_forall_vertices(vi, *nef)
+		{
+			if (f(vi->point())) return;
+		}
+	}
+	else {
+		assert(!"Bad hybrid polyhedron state");
+	}
+}
+
+void CGALHybridPolyhedron::nefPolyBinOp(
+		const std::string &opName, CGALHybridPolyhedron &other,
+		const std::function<void(nef_polyhedron_t &destinationNef, nef_polyhedron_t &otherNef)>
+				&operation)
+{
+	LOG(message_group::Echo, Location::NONE, "", "[fast-csg] %1$s (%2$lu vs. %3$lu facets)",
```

Comment:
I think using the `ECHO: ` prefix in the console for those messages is a bit confusing and may cause problems with people using that marker for scripting OpenSCAD output. With `None` it would just have the `[fast-csg]` prefix which should be enough.

```suggestion
	LOG(message_group::None, Location::NONE, "", "[fast-csg] %1$s (%2$lu vs. %3$lu facets)",
```